test/specs/scale.category.tests.js

```diff
@@ -56,135 +56,206 @@ describe('Category scale tests', function() {
-				data: [10, 5, 0, 25, 78]
-			}],
-			yLabels: ['tick1', 'tick2', 'tick3', 'tick4', 'tick5']
-		};
-
-		var config = Chart.helpers.clone(Chart.scaleService.getScaleDefaults('category'));
-		config.position = 'left'; // y axis
-		var Constructor = Chart.scaleService.getScaleConstructor('category');
-		var scale = new Constructor({
-			ctx: {},
-			options: config,
-			chart: {
-				data: mockData
+		var labels = ['tick1', 'tick2', 'tick3', 'tick4', 'tick5'];
+		var chart = window.acquireChart({
+			type: 'line',
+			data: {
+				datasets: [{
+					xAxisID: 'x',
+					data: [10, 5, 0, 25, 78],
+				}],
+				yLabels: labels
 			},
-			id: scaleID
+			options: {
+				scales: {
+					xAxes: [{
+						id: 'x',
+						type: 'linear'
+					}],
+					yAxes: [{
+						id: 'y',
+						type: 'category',
+					}]
+				}
+			}
 		});
 
-		scale.determineDataLimits();
-		scale.buildTicks();
-		expect(scale.ticks).toEqual(mockData.yLabels);
+		var scale = chart.scales.y;
+		expect(scale.ticks).toEqual(labels);
 	});
 
-	it('Should generate ticks from the axis labels', function() {
+	it ('Should get the correct data label for the index', function() {
 		var labels = ['tick1', 'tick2', 'tick3', 'tick4', 'tick5'];
 		var chart = window.acquireChart({
 			type: 'line',
 			data: {
-				data: [10, 5, 0, 25, 78]
+				datasets: [{
+					data: [10, 5, 0, 25, 78],
+				}],
+				labels: labels
 			},
 			options: {
 				scales: {
 					xAxes: [{
 						id: 'x',
-						type: 'category',
-						labels: labels
+						type: 'category'
 					}]
 				}
 			}
 		});
 
 		var scale = chart.scales.x;
-		expect(scale.ticks).toEqual(labels);
+		expect(scale.getLabelForIndex(1)).toBe('tick2');
 	});
 
-	it ('should get the correct label for the index', function() {
-		var scaleID = 'myScale';
+	it('Should get the correct data xLabel for the index', function() {
+		var labels = ['tick1', 'tick2', 'tick3', 'tick4', 'tick5'];
+		var chart = window.acquireChart({
+			type: 'line',
+			data: {
+				datasets: [{
+					data: [10, 5, 0, 25, 78],
+				}],
+				xLabels: labels
+			},
+			options: {
+				scales: {
+					xAxes: [{
+						id: 'x',
+						type: 'category'
+					}]
+				}
+			}
+		});
 
-		var mockData = {
-			datasets: [{
-				yAxisID: scaleID,
-				data: [10, 5, 0, 25, 78]
-			}],
-			labels: ['tick1', 'tick2', 'tick3', 'tick4', 'tick5']
-		};
+		var scale = chart.scales.x;
+		expect(scale.getLabelForIndex(1)).toBe('tick2');
+	});
 
-		var config = Chart.helpers.clone(Chart.scaleService.getScaleDefaults('category'));
-		var Constructor = Chart.scaleService.getScaleConstructor('category');
-		var scale = new Constructor({
-			ctx: {},
-			options: config,
-			chart: {
-				data: mockData
+	// NOTE: test for vertical line chart
+	// 		 chart type not currently support
+	// it('Should get the correct data yLabel for the index', function() {
+	// 	var labels = ['tick1', 'tick2', 'tick3', 'tick4', 'tick5'];
+	// 	var chart = window.acquireChart({
+	// 		type: 'line',
```

Comment:
@simonbrunel doesn't display as bar chart either. I'm a bit confused right now what to really expect of this?